core/electron-manager/src/backend/ElectronWindowState.ts

```diff
@@ -0,0 +1,118 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { NativeAppStorage } from "@bentley/imodeljs-backend";
+import { StorageValue } from "@bentley/imodeljs-common";
+import { BrowserWindow } from "electron";
+
+// cSpell:ignore unmaximize
+
+/** Maintains main Electron window state for the application.
+ * @beta
+ */
+export class ElectronWindowState {
+  private static _xSettingName = "electron-window-x";
+  private static _ySettingName = "electron-window-y";
+  private static _widthSettingName = "electron-window-width";
+  private static _heightSettingName = "electron-window-height";
+  private static _maximizedSettingName = "electron-window-maximized";
+
+  /** Constructor for ElectronWindowState.
+   * @param storageName       Name used for NativeAppStorage when saving size, position and maximized state
+   * @param defaultWidth      Default width of window
+   * @param defaultHeight     Default height of window
+   * @param defaultMaximized  Default maximized state of window
+   */
+  constructor(public storageName: string, public defaultWidth: number, public defaultHeight: number, public defaultMaximized: boolean = false) {
+  }
+
+  /** Gets the saved window size and position */
+  public getPreviousSizeAndPosition = () => {
+    let setting: StorageValue | undefined;
+    let width = this.defaultWidth;
+    let height = this.defaultHeight;
+    let x: number | undefined;
+    let y: number | undefined;
+
+    const store = NativeAppStorage.open(this.storageName);
+
+    setting = store.getData(ElectronWindowState._widthSettingName);
+    if (setting !== undefined)
+      width = JSON.parse(setting as string);

```

Comment:
why convert everything to/from a string? You can save/get it as a number.

Reply:
Updated

Reply:
i've been making several changes to NativeAppStorage in [this pr](https://github.com/imodeljs/imodeljs/pull/1016). Do you mind if i merge my changes into your PR?

Reply:
That's fine. The values returned from store.getData should be easy to work with. Affan and I chatted about how to make it easier. He suggested getNumber(), getString(), getBoolean(), getUnint8Array(), IsNull(), IsUndefined/exist(key). I went with the type guards. Either way, trusting that you're getting a number or boolean should be easy.